mlrun/utils/notifications/condition_evaluator.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import multiprocessing
+
+import mlrun.model
+from mlrun.utils import logger
+
+from .utils import sanitize_notification
+
+
+def evaluate_notification_condition_in_separate_process(
+    run: mlrun.model.RunObject, notification: mlrun.model.Notification, timeout: int = 5
+):
```

Comment:
I'm thinking this is something we may want to use in many other places, and that it makes sense to make it more generic, accepting a `condition` and a list of `kwargs` (or rather maybe a dict of `key`-> `dict`) to feed into the jinja sandbox?